src/lparser.py

```diff
@@ -0,0 +1,273 @@
+    p[0] = final_out
+
+
+# =========================================================================================================
+# Method Calls
+def p_methodcall(p):  # Define all method calls
+    '''
+    methodcall : resizemethod
+               | addTextmethod
+               | renderVideo
+               | renderGif
+               | cropmethod
+               | addAudiomethod
+               | addExtractedAudiomethod
+               | showVarmethod
+               | showAllVarsmethod
+
+    '''  # TODO add more method calls and implementations.
+    p[0] = p[1]
+
+def p_showVarmethod(p):
+    """
+    Stat method to show he value of a identifier/variable on system.
+
+    showVarmethod : IDENTIFIER
+
+    """
+    print(env[p[1]])
+def p_showAllVarsmethod(p):
+    """
+    Stat method to show all stored variables and identifiers on the system at the moment.
+
+    showAllVarsmethod : SHOW_VARS
+
+    """
+    print(env)
+def p_resizemethod(p):  # Create resize tree.
+    '''
+
+    Method to resize/scale a clip on a identifier/variable by a given number. The resized clip will
+    be permanently changed.
+
+    resizemethod : RESIZE IDENTIFIER BY NUMBER
+    '''
+    final_out = env[p[2]]
+    final_out.resize(new_size=p[4])
+
+
+def p_cropmethod(p):
+    '''
+
+    Method to crop a clip by given aspect_ratios. These ratios are declared as strings and
+    will automatically be executed on call, as per the appropriate ratios defined beforehand.
+    The cropped clip will be permanently changed.
+
+
+    cropmethod : CROP IDENTIFIER BY ASPECT_RATIO
+    '''
+    final_out = env[p[2]]
+    final_out.crop(aspectRatio=p[4])
+
+def p_addAudiomethod(p):
+    '''
+
+    Method to add overlay audio to a clip. If the clip already has audio, it will be overwritten.
+
+    Path must be specified and all 'slashes (/) must be double for the python parser
+    to work and obtain the file.
+
+    The user specifies the path as a string and the time the audio will run (start and end time)
+    taken from the original audio file.
+
+    addAudiomethod : ADD_AUDIO STRING TO IDENTIFIER BETWEEN NUMBER COMMA NUMBER
+    '''
+    final_out = env[p[4]]
+    final_out.addAudioFromFile(audio=p[2],start_time=p[6],end_time=p[8])
+    # p[0] = (p[1],p[2],('var',p[4]t), p[6],p[8])
+def p_addExtractedAudiomethod(p):
+    '''
+
+    Method to add overlay audio to a clip. If the clip already has audio, it will be overwritten.
+
+    The user specifies the audio to extract as a variable/identifier and then the
+    clip to add the extracted audio as another variable/identifier, the time the audio will run (start and end time)
+    taken from the original audio file.
+
+    addExtractedAudiomethod : EXTRACT_AUDIO IDENTIFIER TO IDENTIFIER BETWEEN NUMBER COMMA NUMBER
+
+    '''
+    final_out = env[p[4]]
+    final_out.addAudioFromClip(clipToExtract=p[2], start_time=p[6], end_time=p[8])
+
+def p_addTextmethod(p): # Adds the wanted text to the video or photo

```

Comment:
I will refrain from commenting "same as comment above" for the rest of these methods.

Reply:
It seems the "Run" method has been removed. If so, I would assume all the methods are being parsed as they are received. If this is working for all functionalities without error (not sure where you're saving variables), then it gets a thumbs up from me.

Reply:
As i see, just like before, they are being saved in env, when you make a var assign

Reply:
Thats correct, methods are parsed as they are received/called. As lexdyel says, they are being stored in env global variable.